src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentService.java

```diff
@@ -86,6 +88,29 @@ private void updateDeviceIdsCache( UUID organizationId, UUID deviceEKID, String
         scheduledTasksManager.getDeviceIdsByEKID().put( datasourceId, deviceEKID );
     }
 
+    private Map<UUID, Set<Object>> createDeviceData( String datasourceId, Optional<Datasource> datasource ) {
+        Map<UUID, Set<Object>> data = Maps.newHashMap();
+        data.put( edmCacheManager.getPropertyTypeId( STRING_ID_FQN ), ImmutableSet.of( datasourceId ) );
+
+        if ( datasource.isPresent() ) {
+            if ( AndroidDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
+                AndroidDevice device = (AndroidDevice) datasource.get();
+                data.put( edmCacheManager.getPropertyTypeId( MODEL_FQN ), ImmutableSet.of( device.getModel() ) );
+                data.put( edmCacheManager.getPropertyTypeId( VERSION_FQN ), ImmutableSet.of( device.getOsVersion() ) );
+            }
+
+            if ( IOSDevice.class.isAssignableFrom( datasource.get().getClass() ) ) {
```

Comment:
Shouldn't this be an `else if () {` ?

Reply:
I think the check is still important (robustness and clarity)